src/GraphQL.Tests/Bugs/Bug2159.cs

```diff
@@ -0,0 +1,153 @@
+        [Fact(Skip = "HasArgument currently returns false if GetArgument returns null")]
+        public void HasArgument_NoDefault_SetVariableNull() => AssertQuerySuccess("query($input: Boolean) { hasArgumentNoDefault(arg: $input) }", @"{ ""hasArgumentNoDefault"": true }", "{\"input\":null}".ToInputs());
+
+        //todo: fix HasArgument so it returns false only when no value was supplied
+        [Fact(Skip = "HasArgument currently returns false if GetArgument returns null")]
+        public void HasArgument_WithDefault_None() => AssertQuerySuccess("{ hasArgumentDefault }", @"{ ""hasArgumentWithDefault"": false }");
+
+        //todo: fix HasArgument so it returns false only when no value was supplied
+        [Fact(Skip = "HasArgument currently returns false if GetArgument returns null")]
+        public void HasArgument_WithDefault_UnsetVariable() => AssertQuerySuccess("query($input: Boolean) { hasArgumentWithDefault(arg: $input) }", @"{ ""hasArgumentWithDefault"": false }");
+
+        [Fact]
+        public void HasArgument_WithDefault_Set() => AssertQuerySuccess("{ hasArgumentWithDefault(arg: true) }", @"{ ""hasArgumentWithDefault"": true }");
+
+        [Fact]
+        public void HasArgument_WithDefault_DefaultVariable() => AssertQuerySuccess("query($input: Boolean = true) { hasArgumentWithDefault(arg: $input) }", @"{ ""hasArgumentWithDefault"": true }");
+
+        [Fact]
+        public void HasArgument_WithDefault_SetVariable() => AssertQuerySuccess("query($input: Boolean) { hasArgumentWithDefault(arg: $input) }", @"{ ""hasArgumentWithDefault"": true }", "{\"input\":true}".ToInputs());
+
+        //todo: fix HasArgument so it returns false only when no value was supplied
+        [Fact(Skip = "HasArgument currently returns false if GetArgument returns null")]
+        public void HasArgument_WithDefault_SetNull() => AssertQuerySuccess("{ hasArgumentWithDefault(arg: null) }", @"{ ""hasArgumentWithDefault"": true }");
+
+        [Fact]
+        public void HasArgument_WithDefault_DefaultVariableNull() => AssertQuerySuccess("query($input: Boolean = null) { hasArgumentWithDefault(arg: $input) }", @"{ ""hasArgumentWithDefault"": true }");
+
+        //todo: fix HasArgument so it returns false only when no value was supplied
+        [Fact(Skip = "HasArgument currently returns false if GetArgument returns null")]
+        public void HasArgument_WithDefault_SetVariableNull() => AssertQuerySuccess("query($input: Boolean) { hasArgumentWithDefault(arg: $input) }", @"{ ""hasArgumentWithDefault"": true }", "{\"input\":null}".ToInputs());
+    }
+
+    public class Bug2159Schema : Schema
+    {
+        public Bug2159Schema()
+        {
+            Query = new Bug2159Query();
+        }
+    }
+
+    public class Bug2159Query : ObjectGraphType
+    {
+        public Bug2159Query()
+        {
+            Field<StringGraphType>(
+                "testValue",
+                resolve: ctx => ctx.GetArgument<string>("arg"),
```

Comment:
Could you please swap `arguments` and `resolve` parameters?  It's more common to read.

Reply:
Done